### PR TITLE
using gradle for Jars

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,23 +6,26 @@ USER root
...
-# Install jars to support delta lake spark operations
-ENV HADOOP_AWS_VER=3.3.4
-RUN curl -O https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/${HADOOP_AWS_VER}/hadoop-aws-${HADOOP_AWS_VER}.jar \
-    && mv hadoop-aws-${HADOOP_AWS_VER}.jar /opt/bitnami/spark/jars
+# Install Gradle
+ENV GRADLE_VERSION=8.7
+RUN curl -L https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip -o gradle-bin.zip \
+    && unzip gradle-bin.zip -d /opt \
+    && rm gradle-bin.zip \
+    && ln -s /opt/gradle-${GRADLE_VERSION}/bin/gradle /usr/bin/gradle
```
Comment (MrCreosote) [inline Dockerfile:17]: You should `gradle init` the repo and then use the gradle wrapper file here to install the jars

Comment (MrCreosote) [reply Dockerfile:17]: Check out any of the java services for examples

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+def hadoopAwsVersion = System.getenv('HADOOP_AWS_VER')
+def deltaSparkVersion = System.getenv('DELTA_SPARK_VER')
+def scalaVersion = System.getenv('SCALA_VER')
+
+dependencies {
+    implementation "org.apache.hadoop:hadoop-aws:$hadoopAwsVersion"
+    implementation "io.delta:delta-spark_${scalaVersion}:$deltaSparkVersion"
+    implementation "io.delta:delta-storage:$deltaSparkVersion"
```
Comment (MrCreosote) [inline build.gradle:16]: This is a dependency of delta-spark so not needed

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+task downloadJars(type: Copy) {
+    def jarDir = file('gradle_jars')
+    from configurations.runtimeClasspath
+    into jarDir
+}
```
Comment (MrCreosote) [inline build.gradle:23]: Does this automatically handle dependencies?

File: build.gradle (hunk 2 above)
Comment (Tianhao-Gu) [reply build.gradle]: 👍

File: build.gradle (hunk 3 above)
Comment (Tianhao-Gu) [reply build.gradle:22]: I think so. Not exactly sure though.

File: Dockerfile (hunk 1 above)
Comment (Tianhao-Gu) [reply Dockerfile:17]: run gradle init locally? or with dockerfile? I just need to download the jars not turning this to a java project.

Comment (MrCreosote) [reply Dockerfile:17]: run gradle init in the repo locally, that'll set up gradle correctly with a wrapper file

File: build.gradle (hunk 3 above)
Comment (MrCreosote) [reply build.gradle:22]: If the delta storage jar is still in the jars list it does

File: build.gradle (hunk 2 above)
Comment (MrCreosote) [reply build.gradle]: :+1:

File: Dockerfile (hunk 1 above)
Comment (MrCreosote) [reply Dockerfile:17]: If you're using the wrapper you don't need to install gradle

File: build.gradle (hunk 3 above)
Comment (MrCreosote) [reply build.gradle]: :+1:

File: Dockerfile (hunk 1 above)
Comment (Tianhao-Gu) [reply Dockerfile]: 👍

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**